server/testdriver/junit4/src/main/java/org/infinispan/server/test/junit4/InfinispanXSiteServerTestMethodRule.java

```diff
@@ -0,0 +1,83 @@
+package org.infinispan.server.test.junit4;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.configuration.RestClientConfigurationBuilder;
+import org.infinispan.counter.api.CounterManager;
+import org.infinispan.server.test.api.HotRodTestClientDriver;
+import org.infinispan.server.test.api.RestTestClientDriver;
+import org.infinispan.server.test.api.TestClientXSiteDriver;
+import org.infinispan.server.test.core.TestClient;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import net.spy.memcached.MemcachedClient;
+
+/**
+ * @author Gustavo Lira &lt;glira@redhat.com&gt;
+ * @since 12.0
+ **/
+public class InfinispanXSiteServerTestMethodRule implements TestRule, TestClientXSiteDriver {
+   private Map<String, TestClient> testClients = new HashMap<>();
+
+   public InfinispanXSiteServerTestMethodRule(InfinispanXSiteServerRule serverRule) {
+      Objects.requireNonNull(serverRule, "InfinispanServerRule class Rule is null");
+      serverRule.getTestServers().forEach((k,v) -> this.testClients.put(k, new TestClient(v)));
+   }
+
+   @Override
+   public String getMethodName(String siteName) {
+      return testClients.get(siteName).getMethodName();
+   }
+
+   @Override
+   public HotRodTestClientDriver hotrod(String siteName) {
+      return testClients.get(siteName).hotrod();
+   }
+
+   @Override
+   public RestTestClientDriver rest(String siteName) {
+      return testClients.get(siteName).rest();
+   }
+
+   @Override
+   public CounterManager getCounterManager(String siteName) {
+      return testClients.get(siteName).getCounterManager();
+   }
+
+   // Used for internal test
+   public MemcachedClient getMemcachedClient(String siteName) {
+      return testClients.get(siteName).getMemcachedClient();
+   }
+
+   @Override
+   public Statement apply(Statement base, Description description) {
+      return new Statement() {
+         @Override
+         public void evaluate() throws Throwable {
+            try {
+               testClients.forEach((key, testClient) -> {
+                  testClient.initResources();
+                  testClient.setMethodName(description.getTestClass().getSimpleName() + "." + description.getMethodName());
+               });
```

Comment:
```suggestion
               testClients.values().forEach(TestClient::initResources);
```

Reply:
@pruivo  Without `testClient.setMethodName(description.getTestClass().getSimpleName() + "." + description.getMethodName());` I will need to duplicate the TestClient#getMethodName logic into InfinispanXSiteServerTestMethodRule which IMO is not a good idea, do you have any tips?